<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Rdxplorer2 by rdexplorer</title>
    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/pygment_trac.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
    <script src="javascripts/respond.js"></script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <!--[if lt IE 8]>
    <link rel="stylesheet" href="stylesheets/ie.css">
    <![endif]-->
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">

  </head>
  <body>
      <div id="header">
        <nav>
          <li class="fork"><a href="https://github.com/rdexplorer/RDxplorer2">View On GitHub</a></li>
          <li class="downloads"><a href="https://github.com/rdexplorer/RDxplorer2/zipball/master">ZIP</a></li>
          <li class="downloads"><a href="https://github.com/rdexplorer/RDxplorer2/tarball/master">TAR</a></li>
          <li class="title">DOWNLOADS</li>
        </nav>
      </div><!-- end header -->

    <div class="wrapper">

      <section>
        <div id="title">
          <h1>Rdxplorer2</h1>
          <p>Copy Number Analysis</p>
          <hr>
          <span class="credits left">Project maintained by <a href="https://github.com/rdexplorer">rdexplorer</a></span>
          <span class="credits right">Hosted on GitHub Pages &mdash; Theme by <a href="https://twitter.com/michigangraham">mattgraham</a></span>
        </div>

        <h1>
<a id="rdxplorer2" class="anchor" href="#rdxplorer2" aria-hidden="true"><span class="octicon octicon-link"></span></a>RDxplorer2</h1>

<p>testtest
todo:  goal of prog?</p>

<p><strong>Event Finding:</strong>
Finds the region boundaries of possible copy number events.</p>

<p><strong>Paired Read Extraction:</strong>
Finds the read pairs that have at least one mate 
close to the events found in the previous step.  </p>

<p><strong>Event Validation:</strong>
Assigns probabilities to events based on their read pair info.</p>

<h2>
<a id="installation" class="anchor" href="#installation" aria-hidden="true"><span class="octicon octicon-link"></span></a>Installation</h2>

<p>RDxplorer2 relies on specific versions of the following software:</p>

<ul>
<li><p>samtools version: 0.1.19 or higher
<a href="http://sourceforge.net/projects/samtools/files/samtools/0.1.19/">http://sourceforge.net/projects/samtools/files/samtools/0.1.19/</a></p></li>
<li><p>perl v5.8.8 or higher</p></li>
<li><p>R version 2.15.2 or higher</p></li>
</ul>

<p>Once the prerequisite software is up to date, installation is simple.</p>

<p>Just unzip the archive:</p>

<div class="highlight highlight-bash"><pre>    tar -zxvf rdexplorer-RDxplorer2.tar.gz </pre></div>

<p>And add the root project directory to your path:</p>

<div class="highlight highlight-bash"><pre>    <span class="pl-k">export</span> PATH=<span class="pl-s"><span class="pl-pds">$(</span><span class="pl-c1">pwd</span><span class="pl-pds">)</span></span>/rdexplorer-RDxplorer2:<span class="pl-smi">$PATH</span></pre></div>

<p>Alternatively, you can specify the full path to the 
program scripts when running from the command line.</p>

<h2>
<a id="usage" class="anchor" href="#usage" aria-hidden="true"><span class="octicon octicon-link"></span></a>Usage</h2>

<p>This version of the software allows you to run each part separately.</p>

<p><strong>Event Finding</strong></p>

<p>This perl script takes the location of the bam file as its single command line argument.
The file is assumed to be sorted by position and indexed, with its .bai 
located in the same directory as the .bam.</p>

<p>To run the script:</p>

<div class="highlight highlight-bash"><pre>runEventFinder.pl input.bam</pre></div>

<p>It outputs one summary file(rdxp.chr*.rdx_raw.sum) for each chromosome where events are found.</p>

<p><strong>Paired Read Extraction</strong></p>

<p>There are four input arguments for this script: </p>

<ol>
<li>a sorted and indexed .bam file(with the .bai located the same directory)</li>
<li>one of the following 

<ul>
<li>a summary file containing at least 4 columns for state, chrom, startPos, endPos, 
with events sorted in chromosomal order.</li>
<li>a folder containing the *.sum output files from the first part of the program</li>
</ul>
</li>
<li>a quality score threshold for paired reads.  Reads lower than the threshold this will be ignored.</li>
<li>a quality score threshold for sampling the insert template length distribution.<br>
Reads lower than the threshold will be ignored</li>
</ol>

<p>To run the script:</p>

<ul>
<li>If you have a list of precomputed events:</li>
</ul>

<div class="highlight highlight-bash"><pre>runReadExtractor.sh input.bam summary.txt 32 60</pre></div>

<ul>
<li>If you have a preexisting folder containing the .sum files generated from <strong>event finding</strong>:</li>
</ul>

<div class="highlight highlight-bash"><pre>runReadExtractor.sh input.bam resultFolder 32 60</pre></div>

<ul>
<li>If you have just run <strong>event finding</strong> and are still in the same directory:</li>
</ul>

<div class="highlight highlight-bash"><pre>runReadExtractor.sh input.bam <span class="pl-c1">.</span> 32 60</pre></div>

<p>The output will be a newly created folder called "pairedReads" placed in the current directory.  </p>

<p>Inside are "delEvents.txt" and "dupEvents.txt" with tab delimited columns:</p>

<ul>
<li>cols 1  : a unique identifier for the event</li>
<li>cols 2-4: the 4 columns of interest.</li>
<li>cols 5- : the original summary info.</li>
</ul>

<p>There are 24 paired reads files, "dels_*.pr", and "dups_*.pr", split by chromosome.
Each .pr has the reads' mapping info along with a column showing the event they are from. </p>

<p>There is an insert template length distribution file, "insDistr.txt".<br>
This is a sampling of ~50k template lengths from high quality read pairs.</p>

<p><strong>Event Validation</strong></p>

<p>The program assumes there is a properly constructed "pairedReads" folder in the current directory.
If you have run the previous command, the folder will be loaded with all the necessary files.</p>

<p>To run the script:</p>

<div class="highlight highlight-bash"><pre>  runEventValidator.sh</pre></div>

<p>It will write the event calls into the pairedReads folder.</p>

<p>todo:  add description of del/dup call files.</p>

<h2>
<a id="contributors" class="anchor" href="#contributors" aria-hidden="true"><span class="octicon octicon-link"></span></a>Contributors</h2>

<ul>
<li>Kenny Ye <a href="mailto:kyeblue@gmail.com">kyeblue@gmail.com</a>
</li>
<li>Chris Yoon <a href="mailto:mgflute@gmail.com">mgflute@gmail.com</a>
</li>
<li>Kith Pradhan <a href="mailto:kpradhan@cshl.edu">kpradhan@cshl.edu</a>
</li>
</ul>

<h2>
<a id="license" class="anchor" href="#license" aria-hidden="true"><span class="octicon octicon-link"></span></a>License</h2>
      </section>

    </div>
    <!--[if !IE]><script>fixScale(document);</script><![endif]-->
    
  </body>
</html>