<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">

    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/pygment_trac.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print">

    <title>Rdxplorer2 by rdexplorer</title>
  </head>

  <body>

    <header>
      <div class="container">
        <h1>Rdxplorer2</h1>
        <h2>Copy Number Analysis</h2>

        <section id="downloads">
          <a href="https://github.com/rdexplorer/RDxplorer2/zipball/master" class="btn">Download as .zip</a>
          <a href="https://github.com/rdexplorer/RDxplorer2/tarball/master" class="btn">Download as .tar.gz</a>
          <a href="https://github.com/rdexplorer/RDxplorer2" class="btn btn-github"><span class="icon"></span>View on GitHub</a>
        </section>
      </div>
    </header>

    <div class="container">
      <section id="main_content">
        <h1>
<a id="rdxplorer2" class="anchor" href="#rdxplorer2" aria-hidden="true"><span class="octicon octicon-link"></span></a>RDxplorer2</h1>

<p>todo:  goal of prog?</p>

<p><strong>Event Finding</strong>
Finds the region boundaries of possible copy number events.</p>

<p><strong>Paired Read Extraction</strong>
Finds the read pairs that have at least one mate 
close to the events found in the previous step.  </p>

<p><strong>Event Validation</strong>
Assigns probabilities to events based on their read pair info.</p>

<h2>
<a id="installation" class="anchor" href="#installation" aria-hidden="true"><span class="octicon octicon-link"></span></a>Installation</h2>

<p>RDxplorer2 relies on specific versions of the following software:</p>

<ul>
<li><p>samtools version: 0.1.19 or higher
<a href="http://sourceforge.net/projects/samtools/files/samtools/0.1.19/">http://sourceforge.net/projects/samtools/files/samtools/0.1.19/</a></p></li>
<li><p>perl v5.8.8 or higher</p></li>
<li><p>R version 2.15.2 or higher</p></li>
</ul>

<p>Once the prerequisite software is up to date, installation is simple.</p>

<p>Just unzip the archive:</p>

<div class="highlight highlight-bash"><pre>    tar -zxvf rdexplorer-RDxplorer2.tar.gz </pre></div>

<p>And add the root project directory to your path:</p>

<div class="highlight highlight-bash"><pre>    <span class="pl-k">export</span> PATH=<span class="pl-s"><span class="pl-pds">$(</span><span class="pl-c1">pwd</span><span class="pl-pds">)</span></span>/rdexplorer-RDxplorer2:<span class="pl-smi">$PATH</span></pre></div>

<p>Alternatively, you can specify the full path to the 
program scripts when running from the command line.</p>

<h2>
<a id="usage" class="anchor" href="#usage" aria-hidden="true"><span class="octicon octicon-link"></span></a>Usage</h2>

<p>How to run each part separately.</p>

<p><strong>Event Finding</strong></p>

<p>descrip1</p>

<p><strong>Paired Read Extraction</strong></p>

<p>There are four input arguments for this script: </p>

<ol>
<li>a sorted and indexed .bam file(with the .bai located the same directory)</li>
<li>one of the following 

<ul>
<li>a summary file containing at least 4 columns for state, chrom, startPos, endPos.</li>
<li>a folder containing the *.sum output files from the first part of the program</li>
</ul>
</li>
<li>a quality score threshold for paired reads.  Reads lower than the threshold this will be ignored.</li>
<li>a quality score threshold for sampling the insert template length distribution.<br>
Reads lower than the threshold will be ignored</li>
</ol>

<p>Run the script.</p>

<ul>
<li>If you have a list of precomputed events:</li>
</ul>

<div class="highlight highlight-bash"><pre>runReadExtractor.sh input.bam summary.txt 32 60</pre></div>

<ul>
<li>If you have a folder containing the .sum files generated from the previous step:</li>
</ul>

<div class="highlight highlight-bash"><pre>runReadExtractor.sh input.bam resultFolder 32 60</pre></div>

<p>The output will be a new folder named "pairedReads" created in the directory the script was run from.  </p>

<p>Inside are "delEvents.txt" and "dupEvents.txt" with tab delimited columns:</p>

<ul>
<li>1  : a unique identifier for the event</li>
<li>2-4: the 4 columns of interest.</li>
<li>5- : the original summary info.</li>
</ul>

<p>The 24 paired reads files, "dels_*.pr", and "dups_*.pr", split by chromosome.
Each .pr has the reads' info along with a column showing the event they belong to. </p>

<p>The insert template length distribution file, "insDistr.txt".<br>
This is a sampling of ~50k template lengths from high quality read pairs.</p>

<p><strong>Event Validation</strong></p>

<p>The program assumes there is a properly constructed pairedReads folder in the current directory.</p>

<p>Run the script:</p>

<div class="highlight highlight-bash"><pre>  runEventValidator.sh</pre></div>

<p>It will write the event calls into the pairedReads folder.</p>

<p>todo:  add description of del/dup call files.</p>

<h2>
<a id="contributors" class="anchor" href="#contributors" aria-hidden="true"><span class="octicon octicon-link"></span></a>Contributors</h2>

<ul>
<li>Kenny Ye <a href="mailto:kyeblue@gmail.com">kyeblue@gmail.com</a>
</li>
<li>Chris Yoon <a href="mailto:mgflute@gmail.com">mgflute@gmail.com</a>
</li>
<li>Kith Pradhan <a href="mailto:kpradhan@cshl.edu">kpradhan@cshl.edu</a>
</li>
</ul>

<h2>
<a id="license" class="anchor" href="#license" aria-hidden="true"><span class="octicon octicon-link"></span></a>License</h2>
      </section>
    </div>

    
  </body>
</html>